User controller(valid test)

Sergey113222 opened the pull request:
Hello
This is new branch from last week. In this branch I fixed a lot of little imperfection (type Genre, checked null), update User controller (exception, valid, test), added general sort method  findAllUsers(sortByColumn, sortASC/DESC)   and added repository test

Denisyasyuchenya committed:
Mov-15-16. Please tell if I should add sth

Sergey113222 committed:
Merge remote-tracking branch 'origin/Mov-15-16_valid_user_user-profile' into userController(ValidTest)

# Conflicts:
#	src/main/java/info/movietrash/cinemabase/model/Profile.java
#	src/main/java/info/movietrash/cinemabase/model/User.java

Sergey113222 committed:
16.06.2021 update UserController and RegistrationController
1. add ResponseEntity<>
2. add Validation

Sergey113222 committed:
17.06.2021
1. update UserMovieConverter
2. fix test UserServiceImplTest

Sergey113222 committed:
17.06.2021
1. fix delete method UserController

Sergey113222 committed:
17.06.2021
1. fix delete method UserController(add customn method to repository) and made it faster

Sergey113222 committed:
18.06.2021
1. add findAll UserController with Sorting

Sergey113222 committed:
22.06.21
1. can not add to the table genres externalId from theMovieDB-- have fixed
2. Fix GenreConverterImpl and MovieConverterImpl for saving and mapping genreId[] from table movie_genre

Sergey113222 committed:
23.06.21
1.refactoring
2. fix MovieServiceImplTest

Commit:
Liquibase test config + unit test samples

Sergey113222 committed:
24.06.21
1.update UserController find() with Sort
2. create tests UserConverterImpl.java

Sergey113222 committed:
Merge remote-tracking branch 'origin/Liquibase_test' into userController(ValidTest)

# Conflicts:
#	src/test/java/info/movietrash/cinemabase/converter/impl/UserConverterImplTest.java

Sergey113222 committed:
24.06.21
1.create GenreRepositoryTest test

Sergey113222 committed:
24.06.21
1.add [logs] to .gitignore

Sergey113222 committed:
25.06.21
1. create EmbeddedTestConfig
2. create MovieRepositoryTest
3. create GenreRepositoryTest

4. fix NPE in all Converters

Sergey113222 committed:
27.06.21
1. create UserControllerTest
2. create RegistrationControllerTest
3. refactoring

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/controller/RegistrationController.java:
```diff
@@ -13,13 +18,15 @@
     @PostMapping(value = "/new")
-    public Long register(@RequestBody UserDto userDto) {
-        return userService.createUser(userDto);
+    public ResponseEntity<Long> register(@RequestBody @Valid UserDto userDto) {
```
у тебя @RestController - не нужно оборачивать в REsponseEntity - по умолчанию твой ответ будет преобразован в json

Sergey113222 replied:
solved

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/controller/RegistrationController.java:
```diff
@@ -13,13 +18,15 @@
 
     @PutMapping(value = "/update")
-    public void updateUser(@RequestBody UserDto userDto) {
+    public ResponseEntity<UserDto> updateUser(@RequestBody @Valid UserDto userDto) {
```
аналгично

Sergey113222 replied:
solved

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/controller/UserController.java:
```diff
@@ -14,13 +16,23 @@
+            @RequestParam String sortDirection,
+            @RequestParam String sortColumn) {
+        List<UserDto> userDtoList = userService.findAllUsers(sortDirection, sortColumn);
+        return ResponseEntity.ok().body(userDtoList);
```
тоже самое

Sergey113222 replied:
solved

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/controller/UserController.java:
```diff
@@ -14,13 +16,23 @@
-    public void deleteUserById(@PathVariable("id") Long id) {
+    public ResponseEntity<UserDto> deleteUserById(@PathVariable("id") @Min(1) Long id) {
         userService.deleteUser(id);
+        return ResponseEntity.noContent().build();
```
тоже самое

Sergey113222 replied:
solved

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/controller/UserMovieController.java:
```diff
@@ -31,7 +31,7 @@
 
     @PutMapping
-    public ResponseEntity<MovieDto> updateFavouriteMovie(@RequestBody @Min(1) MovieDto movieDto, @RequestParam @Valid Long userMovieId) {
+    public ResponseEntity<MovieDto> updateFavouriteMovie(@RequestBody @Valid MovieDto movieDto, @RequestParam @Min(1) Long userMovieId) {
```
тоже самое

RomanKashenok replied:
и убери наверное этот @Min(1) чет он везде какой-то лишний

Sergey113222 replied:
solved
deleted

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/converter/impl/DirectionConverterImpl.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Override
+    public Sort.Direction replaceStringThroughDirection(String sortDirection) {
+        if (sortDirection.equalsIgnoreCase("DESC")) {
```
сделай выбор направления прямо в enum, не используй строки тут

Sergey113222 replied:
Ты имеешь в ввиду как в этом примере ?
List<Passenger> passengers = repository.findAll(Sort.by(Sort.Direction.ASC, "seatNumber"));
Просто я хочу со стороны пользователя хочу управлять направлением сортировки и я же не могу в контроллере передать Direction.ASC вот поэтому и использую стрингу

Sergey113222 replied:
!!!!!
01.07.21
fix without String and DirectionConverterImpl.java, only with enum Sort.Direction direction

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/converter/impl/MovieConverterImpl.java:
```diff
@@ -18,12 +32,23 @@ public Movie toModel(@NonNull MovieDto movieDto) {
+
+        List<Long> genreExternalIds = movieDto.getGenreIds();
+        List<Genre> genres = new ArrayList<>();
+        for (Long genreExternalId : genreExternalIds) {
```
сделай поиск жанров не перебором по одному, а поиск сразу всех (select ... where genre_id in (...) )

Sergey113222 replied:
@Query("select g from Genre g where g.externalId IN :genreExternalIds")
    List<Genre> findAllByExternalId(@Param("genreExternalIds") List<Long> genreExternalIds);

ох уж это быстродействие усложняет людям жизнь - выполнено

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/converter/impl/UserMovieConverterImpl.java:
```diff
@@ -0,0 +1,54 @@
+        movieDto.setAdult(userMovie.getMovie().getAdult());
+
+        List<Long> genresIds = new ArrayList<>();
+        for (Genre genre : userMovie.getMovie().getGenres()) {
```
давай stream() заюзаем

Sergey113222 replied:
stream+

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/dto/ProfileDto.java:
```diff
@@ -4,17 +4,33 @@
+            = "About must be between 10 and 200 characters")
     private String about;
+    @Email(message = "Email should be valid")
+    @Pattern(regexp = ".+@.+\\.[a-z]+")
```
а за чем тебе паттерн, если уже есть @Email ?

Sergey113222 replied:
only @Email

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/dto/UserDto.java:
```diff
@@ -5,17 +5,26 @@
     private String username;
+    @NotEmpty
+    @NotNull
+    @Size(min = 6, max = 50, message = "Password must be between 6 and 50 characters")
```
выбери одну какую-нить аннотацию. Size - уже говорит что не пустое поле определенного размера

Sergey113222 replied:
left only @Size

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/repository/UserRepository.java:
```diff
@@ -1,14 +1,23 @@
 
+    @Modifying
+    @Query("update User u set u.active = false where u.id = :id")
+    void updateUser(@Param("id") Long id);
```
наверное это все же remove/delete , но не updateUser

Sergey113222 replied:
yes
renamed deleteUser

RomanKashenok reviewed (changes requested):
Минимально поменяй плиз

Sergey113222 committed:
30.06.21
fix pull request mistakes

Sergey113222 committed:
30.06.21
fix pull request mistakes 2

Sergey113222 committed:
01.07.21
fix without DirectionConverterImpl.java, only with enum Sort.Direction direction

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/converter/impl/MovieConverterImpl.java:
```diff
@@ -32,7 +51,14 @@ public MovieDto toDto(Movie movie) {
+        List<Genre> genres = movie.getGenres();
+        List<Long> genresIds = new ArrayList<>();
+        for (Genre genre : genres) {
+            genresIds.add(genre.getExternalId());
```
stream is better

Sergey113222 replied:
done

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/converter/impl/UserMovieConverterImpl.java:
```diff
@@ -0,0 +1,51 @@
+
+    @Override
+    public UserMovie createUserMovie(User user, Movie movie, MovieDto movieDto) {
+        if ((user == null) || (movie == null) || (movieDto == null)) {
```
allNotNull(Object... values) is better

Sergey113222 replied:
if (ObjectUtils.allNotNull(user, movie, movieDto))---- done

RomanKashenok commented on src/main/java/info/movietrash/cinemabase/service/impl/UserServiceImpl.java:
```diff
@@ -1,52 +1,61 @@
-                new IllegalArgumentException(String.format(ErrorMessages.RESOURCE_NOT_FOUND, User.class, id)));
-        UserDto userDto = userConverter.toDto(user);
-        return userDto;
+                new ResourceNotFoundException(String.format(ErrorMessages.RESOURCE_NOT_FOUND, id)));
```
IllegalArgumentException is better, or better custom exception (ServiceExeption)

Sergey113222 replied:
ok

RomanKashenok reviewed (approved):
will some comments

Sergey113222 committed:
04.07.21
fix Roma's little mistakes to merge into development
